src/main/java/appeng/parts/PartPlacement.java

```diff
@@ -336,11 +299,6 @@ else if( host != null && !host.canAddPart( held, AEPartLocation.fromFacing( side
 				final Block blkID = world.getBlockState( te_pos ).getBlock();
 				tile = world.getTileEntity( te_pos );
 
-				if( tile != null && IntegrationRegistry.INSTANCE.isEnabled( IntegrationType.FMP ) )
-				{
-					host = ( (IFMP) IntegrationRegistry.INSTANCE.getInstance( IntegrationType.FMP ) ).getOrCreateHost( tile );
-				}
-
 				if( ( blkID == null || blkID.isReplaceable( world, te_pos ) || host != null ) ) // /&& side !=
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Redundant nullcheck of host, which is known to be non-null in appeng.parts.PartPlacement.place(ItemStack, BlockPos, EnumFacing, EntityPlayer, EnumHand, World, PartPlacement$PlaceType, int) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)